File: mobile/lib/utils/error_helper.dart
import 'package:dio/dio.dart';

/// Convertit une exception en message lisible pour l'utilisateur.
String mapErrorToMessage(Object e, [String fallback = 'Une erreur inattendue s\'est produite.']) {
  if (e is DioException) {
    // Message personnalisé injecté par l'intercepteur (401/403)
    if (e.message != null && e.message!.isNotEmpty && e.type == DioExceptionType.badResponse) {
      final serverMsg = e.response?.data?['message'];
      if (serverMsg is String && serverMsg.isNotEmpty) return serverMsg;
      return e.message!;
    }

    return switch (e.type) {
      DioExceptionType.connectionTimeout ||
      DioExceptionType.sendTimeout ||
      DioExceptionType.receiveTimeout =>
        'La connexion est trop lente. Vérifiez votre réseau.',
      DioExceptionType.connectionError =>
        'Impossible de contacter le serveur. Vérifiez votre connexion.',
      DioExceptionType.badResponse => () {
          final serverMsg = e.response?.data?['message'];
          if (serverMsg is String && serverMsg.isNotEmpty) return serverMsg;
          return 'Erreur serveur (${e.response?.statusCode ?? '?'}).';
        }(),
      DioExceptionType.cancel => 'Requête annulée.',
      _ => 'Erreur réseau. Vérifiez votre connexion.',
    };
  }

  return fallback;
}

File: front/src/api/errorHelper.js
/**
 * Extrait un message lisible depuis une erreur Axios.
 * Priorité : validation → message serveur → erreur réseau/timeout → fallback.
 */
export const extractErrorMessage = (err, fallback = 'Une erreur est survenue.') => {
    if (err?.response?.data?.errors) {
        return Object.values(err.response.data.errors).flat().join(' ');
    }
    if (err?.response?.data?.message) {
        return err.response.data.message;
    }
    if (err?._userMessage) {
        return err._userMessage;
    }
    if (!err?.response) {
        return 'Impossible de contacter le serveur. Vérifiez votre connexion.';
    }
    return fallback;
};
